annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/BaseAnnotationValidator.java

```diff
@@ -486,11 +487,18 @@ protected void validateSubtitleOrCaption(Annotation webAnnotation)
 	// validate body
 	Body body = webAnnotation.getBody();
 	validateFullTextResource(body);
-	// check mandatory field edmRights
+	// check mandatory field body.format (please note that this field is saved in the "contentType" field of the given Resource)
 	if (StringUtils.isBlank(body.getContentType())) {
 	    throw new PropertyValidationException(I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD,
 		    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD, new String[] { "subtitle.body.format" });
 	}
+	// check if the body.format field has a valid value
+	String[] validSubtitlesFormats = getConfiguration().getAnnotationSubtitlesFormats().split(",");
```

Comment:
this processing of the formats string should happen in Configuration class and the class should return a Set<String>, that that the verification of the format is straight forward and performant

Reply:
Improved.